apps/bfd-model/bfd-model-dsl-codegen/bfd-model-dsl-codegen-plugin/src/main/java/gov/cms/model/dsl/codegen/plugin/model/ColumnBean.java

```diff
@@ -0,0 +1,356 @@
+package gov.cms.model.dsl.codegen.plugin.model;
+
+import com.google.common.base.Strings;
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.TypeName;
+import java.math.BigDecimal;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.persistence.GenerationType;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+/** Model object for columns in a SQL database mapping. */
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class ColumnBean {
+  /** Regex used to recognize numeric columns by their SQL type. */
+  private static final Pattern NumericTypeRegex =
+      Pattern.compile("(numeric|decimal)\\((\\d+)(,(\\d+))?\\)", Pattern.CASE_INSENSITIVE);
+  /** Group number in {@link NumericTypeRegex} that contains the numeric precision value. */
+  private static final int NumericPrecisionGroup = 2;
+  /** Group number in {@link NumericTypeRegex} that contains the numeric scale value. */
+  private static final int NumericScaleGroup = 4;
+  /**
+   * Regex to extract the length from a SQL type if it is character type with a defined integer
+   * length.
+   */
+  private static final Pattern CharacterLengthRegex =
+      Pattern.compile("char\\((\\d+)\\)", Pattern.CASE_INSENSITIVE);
+  /** Group number in {@link CharacterLengthRegex} that contains the numeric length value. */
+  private static final int CharacterLengthGroup = 1;
+  /** Regex used to recognize date columns by their SQL type. */
+  private static final Pattern DateTypeRegex = Pattern.compile("date", Pattern.CASE_INSENSITIVE);
+
+  /** Name of the field in the entity object corresponding to this column. */
+  private String name;
+  /** Alternative name used in database for this column. Defaults to {@link name}. */
+  private String dbName;
+  /** SQL database type for this column. */
+  private String sqlType;
+  /** Specific java type for the field in the entity object corresponding to this column. */
+  private String javaType;
+  /**
+   * Alternative type for generated accessor methods (getter/setter) for the field in the entity
+   * object corresponding to this column.
+   */
+  private String javaAccessorType;
+  /**
+   * Name of an enum type defined in the same mapping as this column. Should be found by calling
+   * {@link MappingBean#findEnum(String)} on the mapping.
+   */
+  private String enumType;
+  /** Text for insertion into the generated entity as a javadoc comment on the field. */
+  private String comment;
+  /** Indicates whether the column in the database is nullable. */
+  private boolean nullable = true;
+  /** Indicates whether the column in the database is an {@link GenerationType.IDENTITY} column. */
```

Comment:
I believe javadoc uses `#` separators (i.e. `{@link GenerationType#IDENTITY}`) here and other places.